x-pack/plugins/security_solution/public/management/pages/policy/view/event_filters/flyout/policy_event_filters_flyout.tsx

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { isEmpty, without } from 'lodash/fp';
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import {
+  EuiTitle,
+  EuiFlyout,
+  EuiSpacer,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiCallOut,
+  EuiEmptyPrompt,
+} from '@elastic/eui';
+import { SearchExceptions } from '../../../../../components/search_exceptions';
+import { ImmutableObject, PolicyData } from '../../../../../../../common/endpoint/types';
+import { useToasts } from '../../../../../../common/lib/kibana';
+import { useSearchNotAssignedEventFilters, useBulkUpdateEventFilters } from '../hooks';
+import { PolicyArtifactsAssignableList } from '../../artifacts/assignable';
+
+interface PolicyEventFiltersFlyoutProps {
+  policyItem: ImmutableObject<PolicyData>;
+  onClose: () => void;
+}
+
+const MAX_ALLOWED_RESULTS = 100;
+
+export const PolicyEventFiltersFlyout = React.memo<PolicyEventFiltersFlyoutProps>(
+  ({ policyItem, onClose }) => {
+    const toasts = useToasts();
+    const [selectedArtifactIds, setSelectedArtifactIds] = useState<string[]>([]);
+    const [currentFilter, setCurrentFilter] = useState<string>('');
+
+    const bulkUpdateMutation = useBulkUpdateEventFilters({
+      onUpdateSuccess: (updatedExceptions: ExceptionListItemSchema[]) => {
```

Comment:
I have not worked with react query, so I'm not sure about this - but should these all be memoized? are they triggering unnecessary react render cycles by having new instances created everytime?

Reply:
No, internally react-query is already taking care of this.